### PR TITLE
Add tests for Task 03

### PR DESCRIPTION
**Pull request by NickyMateev**
PR to add additional tests covering some corner cases.

### CODE REVIEW

**Comment by ironsmile** [comment]
Благодаря за новите тестове, но със сигурност няма да имам време да ги прегледам преди крайния срок за домашното.

File: tasks/03/sample_test.go [hunk 1]
```diff
@@ -18,6 +18,54 @@ func TestSampleSimpleOperations(t *testing.T) {
...
+	a, b, c := geom.NewVector(0, -2, 0), geom.NewVector(3, 0, 0), geom.NewVector(0, 1, 0)
+	prim = NewTriangle(a, b, c)
+	ray := geom.NewRay(geom.NewVector(0, 0, 3), geom.NewVector(0, 0, -2))
+
+	if !prim.Intersect(ray) {
+		t.Errorf("Expected ray %#v to intersect triangle %#v but it did not.", ray, prim)
+	}
+}
+
+func TestSampleSimpleSphereShouldIntersect(t *testing.T) {
```
Comment (MVrachev) [inline tasks/03/sample_test.go]: Този тест не е верен.
Картинката изглежда така: 
image

А ни е казано, че t >= 0: 



Тоест лъча НЕ трябва да пресича сферата.

**Review by MVrachev — CHANGES_REQUESTED** [review]

File: tasks/03/sample_test.go (hunk 1 above)
Comment (ironsmile) [reply tasks/03/sample_test.go]: Я какъв хубав инструмент за визуализация. Как се казва?

Comment (MVrachev) [reply tasks/03/sample_test.go]: https://www.geogebra.org/classic/3d

Comment (NickyMateev) [reply tasks/03/sample_test.go]: Good catch 👍 
Тук очевидно съм проспал рестрикцията за t.

Comment (ironsmile) [reply tasks/03/sample_test.go]: Мерси! :)
